src/test/java/org/opentripplanner/transit/raptor/moduletests/D01_TimeDependentAccessTest.java

```diff
@@ -0,0 +1,241 @@
+                );
+
+        var response = raptorService.route(requestBuilder.build(), data);
+
+        assertEquals(
+                ""
+                        + "Walk 2m ~ 2 ~ BUS R1 0:15 0:30 ~ 5 ~ Walk 1m [00:13:00 00:31:00 18m, cost: 2220]\n"
+                        + "Walk 2m ~ 2 ~ BUS R2 0:20 0:35 ~ 5 ~ Walk 1m [00:18:00 00:36:00 18m, cost: 2220]\n"
+                        + "Walk 2m ~ 2 ~ BUS R3 0:25 0:40 ~ 5 ~ Walk 1m [00:20:00 00:41:00 21m, cost: 2220]",
+                pathsToString(response)
+        );
+    }
+
+    /**
+     * The access is only open after 00:18 and before 00:22. This means that we have to leave after
+     * 00:18 but before 00:20 (~ arrive at the stop at 00:20 at the earliest and 00:22 at the
+     * latest). Due to this: - boarding R1 is not possible - there needs to be a wait of 3 minutes
+     * for R3 - R4 is discarded since it is not better than taking R3
+     */
+    @Test
+    public void partiallyOpenIntervalTest() {
+        requestBuilder.profile(RaptorProfile.MULTI_CRITERIA)
+                .searchParams()
+                .addAccessPaths(
+                        new TimeDependentAccessEgress(
+                                STOP_2, D2m, null, hm2time(0, 18), hm2time(0, 22))
+                );
+
+        var response = raptorService.route(requestBuilder.build(), data);
+
+        assertEquals(
+                ""
+                        + "Walk 2m ~ 2 ~ BUS R2 0:20 0:35 ~ 5 ~ Walk 1m [00:18:00 00:36:00 18m, cost: 2220]\n"
+                        + "Walk 2m ~ 2 ~ BUS R3 0:25 0:40 ~ 5 ~ Walk 1m [00:20:00 00:41:00 21m, cost: 2220]",
+                pathsToString(response)
+        );
+    }
+
+    /**
+     * If AccessEgress#earliestDepartureTime returns -1, than: - openInFirstHalfIntervalTest,
+     * partiallyOpenIntervalTest don't wait at the stop - this test correctly returns an empty
+     * result.
+     *
+     * If AccessEgress#earliestDepartureTime returns `closing` time instead of `-1`, than: -
+     * openInFirstHalfIntervalTest, partiallyOpenIntervalTest correctly wait at the stop - this test
+     * incorrectly returns a trip plan before the requested departure
+     */
+    @Test
+    public void tooEarlyDeparture() {
+        requestBuilder.profile(RaptorProfile.MULTI_CRITERIA)
+                .searchParams()
+                .earliestDepartureTime(T00_30)
+                .addAccessPaths(
+                        new TimeDependentAccessEgress(STOP_2, D2m, null, T00_00, hm2time(0, 22))
+                );
+
+        var response = raptorService.route(requestBuilder.build(), data);
+
+        assertEquals(
+                ""
+                        + "Walk 2m ~ 2 ~ BUS R3 0:25 0:40 ~ 5 ~ Walk 1m [00:20:00 00:41:00 21m, cost: 2220]",
+                pathsToString(response)
+        );
+    }
+
+    private static class TimeDependentAccessEgress extends AccessEgress {
+
+        private final int opening;
+        private final int closing;
```

Comment:
I think this can be integrated into  the TestTransfer class, then we can use it for testing  FLEX as well and it will be less code in this test.

Reply:
Good idea, handling _opening/closing_ is now part of `TestTransfer`.